src/value.cpp

```diff
@@ -56,145 +60,100 @@ void value::set(tag&& t)
 }
 
 //Primitive assignment
-//FIXME: Make this less copypaste!
-value& value::operator=(int8_t val)
+namespace // helper functions local to this translation unit
 {
-    if(!tag_)
-        set(tag_byte(val));
-    else switch(tag_->get_type())
+    template<typename T>
+    std::unique_ptr<tag> make_numeric_tag(tag_type type, T val)
```

Comment:
can you move this near std::unique_ptr<tag> tag::create
maybe even name it the same but make it have more params so you could create the tag with a value directly